File: src/test/java/com/morak/performancetracker/aop/rest/RestAdvisorTest.java
package com.morak.performancetracker.aop.rest;

import static org.assertj.core.api.Assertions.assertThat;

import com.morak.performancetracker.context.Accumulator;
import com.morak.performancetracker.context.ResultMapper;
import org.junit.jupiter.api.Test;

class RestAdvisorTest {

    @Test
    void advisor를_생성한다() {
        // given
        RestPointcut pointcut = new RestPointcut();
        RestAdvice advice = new RestAdvice(new RestMonitor(), new Accumulator(new ResultMapper()));
        // when
        RestAdvisor advisor = new RestAdvisor(pointcut, advice);
        // then
        assertThat(advisor).isNotNull();
    }
}

File: src/test/kotlin/com/morak/performancetracker/aop/rest/RestAdvisorTest.kt
package com.morak.performancetracker.aop.rest

import com.morak.performancetracker.context.Accumulator
import com.morak.performancetracker.context.ResultMapper
import org.assertj.core.api.Assertions
import org.junit.jupiter.api.Test

internal class RestAdvisorTest {
    @Test
    fun advisor를_생성한다() {
        // given
        val pointcut = RestPointcut()
        val advice = RestAdvice(RestMonitor(), Accumulator(ResultMapper()))
        // when
        val advisor = RestAdvisor(pointcut, advice)
        // then
        Assertions.assertThat(advisor).isNotNull()
    }
}
